src/Neo/Plugins/Plugin.cs

```diff
@@ -168,47 +137,56 @@ protected IConfigurationSection GetConfiguration()
                 .GetSection("PluginConfiguration");
         }
 
-        private static void LoadPlugin(Assembly assembly)
+        internal static void LoadPlugins()
         {
-            foreach (var type in assembly.ExportedTypes)
+            if (Directory.Exists(PluginsDirectory) == false)
+                return;
+
+            var pluginDirs = Directory.GetDirectories(PluginsDirectory);
+            var pluginAssemblyContext = new PluginAssemblyLoadContext(pluginDirs);
+
+            foreach (var pluginPath in pluginDirs)
             {
-                if (!type.IsSubclassOf(typeof(Plugin))) continue;
-                if (type.IsAbstract) continue;
+                var pluginName = GetFileName(pluginPath);
+                var pluginFileName = Combine(pluginPath, $"{pluginName}.dll");
 
-                var constructor = type.GetConstructor(Type.EmptyTypes);
-                if (constructor == null) continue;
+                if (File.Exists(pluginFileName) == false)
+                    continue;
 
-                try
+                // Provides isolated, dynamic loading and unloading of assemblies and
+                // their dependencies. Each ALC instance manages the resolution and
+                // loading of assemblies and supports loading multiple versions of the
+                // same assembly within a process by isolating them in different contexts.
+                var assemblyName = new AssemblyName(pluginName);
+                var pluginAssembly = pluginAssemblyContext.LoadFromAssemblyName(assemblyName);
+
+                var neoPluginClassType = pluginAssembly.ExportedTypes
+                    .FirstOrDefault(
+                        static f =>
+                            f.IsAssignableTo(typeof(Plugin)) && f.IsAbstract == false
+                    );
+
+                if (neoPluginClassType is null)
+                    pluginAssemblyContext.Unload();
```

Comment:
The context it's shared, but if only one fail, all are unloaded
Steps to reproduce:
- Copy regular dll in plugin directory
- All plugins will be disabled